regress/pesterTests/SSH.Tests.ps1

```diff
@@ -1,78 +1,56 @@
-        $port = $OpenSSHTestInfo["Port"]
-        $ssouser = $OpenSSHTestInfo["SSOUser"]
-
-        $testDir = Join-Path $OpenSSHTestInfo["TestDataPath"] $suite
+        $port = 47002
+        $testDir = Join-Path $env:temp $suite
         if(-not (Test-Path $testDir))
         {
-            $null = New-Item $testDir -ItemType directory -Force -ErrorAction SilentlyContinue
-        }
-        $acl = Get-Acl $testDir
-        $rights = [System.Security.AccessControl.FileSystemRights]"Read, Write"
-        $accessRule = New-Object System.Security.AccessControl.FileSystemAccessRule($ssouser, $rights, "ContainerInherit,Objectinherit", "None", "Allow")
-        $acl.SetAccessRule($accessRule)
-        Set-Acl -Path $testDir -AclObject $acl
-        $platform = Get-Platform
-        #skip on ps 2 becase non-interactive cmd require a ENTER before it returns on ps2
-        $skip = ($platform -eq [PlatformType]::Windows) -and ($PSVersionTable.PSVersion.Major -le 2)
-
-        <#$testData = @(
-            @{
-                Title = 'Simple logon no option';                
-                LogonStr = "$($server.localAdminUserName)@$($server.MachineName)"
-                Options = ""
-            },
-            @{
-                Title = 'Simple logon using -C -l option'
-                LogonStr = $server.MachineName
-                Options = "-C -l $($server.localAdminUserName)"
-            }
-        )
+            New-Item $testDir -ItemType directory -Force -ErrorAction SilentlyContinue | Out-Null
+        }
+        else {
+            Get-ChildItem $testDir | Remove-Item | Out-Null
+        }
+
+        $Host_Key_File = "$testDir\hostkey_ed25519"
         
-        $testData1 = @(
-            @{
-                Title = "logon using -i -q option"
-                LogonStr = "$($server.localAdminUserName)@$($server.MachineName)"
-                Options = '-i $identifyFile -q'
-            },
-            @{
-                Title = "logon using -i option"
-                LogonStr = "$($server.localAdminUserName)@$($server.MachineName)"
-                Options = '-i $identifyFile'
-            },
-            @{
-                Title = "logon using -i -c  option"
-                LogonStr = "$($server.localAdminUserName)@$($server.MachineName)"
-                Options = '-i $identifyFile -c aes256-ctr'
-            },
-             -V does not redirect to file
-            @{
-                Title = "logon using -i -V option"
-                LogonStr = "$($server.localAdminUserName)@$($server.MachineName)"
-                Options = '-i $identifyFile -V'
-                SkipVerification = $true
-            },
-            @{
-                Title = 'logon using -i -l option'
-                LogonStr = $server.MachineName
-                Options = '-i $identifyFile -l $($server.localAdminUserName)'
-            }
-        )#>
+        & "$binpath\ssh-keygen.exe" -t ed25519 -P "`"`"" -f $Host_Key_File
```

Comment:
Since these are common for most of the test suites, we could make this into a utility helper in common.ps1 and share. Perhaps call it common_setup.

Reply:
I thought about it. but I can't completely hide these steps, the ssh client needs host info to build known_host, server setup needs the user pub key to build authorized_keys. If we put them all in common_setup, I will have them as output parameters then.